src/inference/src/dev/iremote_context.cpp

```diff
@@ -9,3 +9,7 @@
 ov::SoPtr<ov::ITensor> ov::IRemoteContext::create_host_tensor(const ov::element::Type type, const ov::Shape& shape) {
     return ov::SoPtr<ov::ITensor>(ov::make_tensor(type, shape), nullptr);
 }
+
+ov::SoPtr<ov::ITensor> ov::IRemoteContext::create_user_tensor(const ov::element::Type type, const ov::Shape& shape) {
+    return create_host_tensor(type, shape);
+}
```

Comment:
This is a public API, and I think we can avoid introducing new function just for fixing types conversion problem

How about letting RemoteTensorImpl allocate requested data type w/o implicit data types conversion inside its c-tr? Probably, it will also fix ambiguous behavior in some other cases (e.g. in case of remote tensors). And then you just need to carefully convert types where it is really needed and use original type otherwise.